### PR TITLE
Solve missing Serializers exception

### LINKED ISSUE #462: Zipline-Cli fails on ZiplineManifest not being @serializable (though it is)
```
kotlinx.serialization.SerializationException: Serializer for class 'ZiplineManifest' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
        at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
        at kotlinx.serialization.internal.PlatformKt.platformSpecificSerializerNotRegistered(Platform.kt:29)
        at kotlinx.serialization.SerializersKt__SerializersKt.serializer(Serializers.kt:60)
        at kotlinx.serialization.SerializersKt.serializer(Unknown Source)
        at app.cash.zipline.loader.ZiplineDownloader.downloadZiplineManifest(ZiplineDownloader.kt:112)
        at app.cash.zipline.loader.ZiplineDownloader.access$downloadZiplineManifest(ZiplineDownloader.kt:30)
        at app.cash.zipline.loader.ZiplineDownloader$downloadZiplineManifest$1.invokeSuspend(ZiplineDownloader.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:281)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:87)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:61)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:40)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at app.cash.zipline.cli.Download.download(Download.kt:57)
        at app.cash.zipline.cli.Download.run(Download.kt:63)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at app.cash.zipline.cli.Main$Companion.main(Main.kt:70)
        at app.cash.zipline.cli.Main.main(Main.kt)

```

### PR DESCRIPTION
**Pull request by adrw**
[CLOSES #462]

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]
No tests are hitting this codepath?

File: zipline-loader/src/jvmTest/kotlin/app/cash/zipline/loader/ZiplineDownloaderTest.kt [hunk 1]
```diff
@@ -91,4 +93,34 @@ class ZiplineDownloaderTest {
...
+    assertFalse(fileSystem.exists(downloadDir / testFixturesJvm.bravoSha256Hex))
+
+    val baseUrl = "http://10.0.2.2:8080/".toHttpUrl()
+    downloader = ZiplineDownloader(
+      dispatcher = dispatcher,
+      httpClient = OkHttpZiplineHttpClient(baseUrl, OkHttpClient()),
+      downloadDir = downloadDir,
+      downloadFileSystem = fileSystem,
+    )
+    downloader.download("https://treelot.sqprod.co/latest/treehouse-activity/manifest.zipline.json")
```
Comment (JakeWharton) [inline zipline-loader/src/jvmTest/kotlin/app/cash/zipline/loader/ZiplineDownloaderTest.kt]: Needs to be served via MockWebServer and not hit the real internet

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineDownloader.kt [hunk 2]
```diff
@@ -66,6 +66,9 @@ class ZiplineDownloader(
     val manifestByteString = concurrentDownloadsSemaphore.withPermit {
       httpClient.download(manifestUrl)
     }
+
+    // We have to manually pass in the serializer here to let ZiplineDownloader be used in Zipline-CLI
+    //  because GraalVM doesn't support the reflection for the more magic extension function.
```
Comment (JakeWharton) [inline zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineDownloader.kt]: Being explicit here is better regardless of this use.

**Review by JakeWharton — APPROVED** [review]

File: zipline-loader/src/jvmTest/kotlin/app/cash/zipline/loader/ZiplineDownloaderTest.kt (hunk 1 above)
Comment (adrw) [reply zipline-loader/src/jvmTest/kotlin/app/cash/zipline/loader/ZiplineDownloaderTest.kt]: Thanks, I'll set up a test that does that

**Comment by adrw** [comment]
There weren't any tests that exercised use of the CLI because I naively thought that since the Command classes couldn't take constructor parameters that tests weren't possible. With further thought, MockWebServer + using a /tmp/ real FS directory should work.

**Comment by adrw** [comment]
Waiting for this to land to use common test fixtures https://github.com/cashapp/zipline/pull/466

File: zipline-cli/src/test/kotlin/app/cash/zipline/cli/DownloadTest.kt [hunk 3]
```diff
@@ -40,6 +71,53 @@ class DownloadTest {
     assertEquals("Missing required option: '--download-dir=<downloadDir>'", exception.message)
   }
 
+  @Test fun downloadFromMockWebServer() {
+    // Wipe and re-create local test download directory
+    val downloadDirPath = "/tmp/zipline/download".toPath()
```
Comment (swankjesse) [inline zipline-cli/src/test/kotlin/app/cash/zipline/cli/DownloadTest.kt]: Also consider `FileSystem.SYSTEM_TEMPORARY_DIRECTORY`

File: zipline-loader-testing/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+plugins {
+  kotlin("multiplatform")
+  kotlin("plugin.serialization")
+}
+
+kotlin {
+  jvm()
+
+  js {
```
Comment (swankjesse) [inline zipline-loader-testing/build.gradle.kts]: I don’t think the loader should support running inside JS

**Review by swankjesse — APPROVED** [review]

**Pull request merged**